sentinelhub/data_sources.py

```diff
@@ -0,0 +1,522 @@
+        sensor_type=_SensorType.EW,
+        polarization=_Polarization.DH,
+        resolution=_Resolution.MEDIUM,
+        bands=_Bands.SENTINEL1_EW
+    )
+    SENTINEL1_EW_ASC = SENTINEL1_EW.derive(
+        orbit_direction=OrbitDirection.ASCENDING
+    )
+    SENTINEL1_EW_DES = SENTINEL1_EW.derive(
+        orbit_direction=OrbitDirection.DESCENDING
+    )
+    SENTINEL1_EW_SH = SENTINEL1_EW.derive(
+        polarization=_Polarization.SH,
+        bands=_Bands.SENTINEL1_EW_SH
+    )
+    SENTINEL1_EW_SH_ASC = SENTINEL1_EW_SH.derive(
+        orbit_direction=OrbitDirection.ASCENDING
+    )
+    SENTINEL1_EW_SH_DES = SENTINEL1_EW_SH.derive(
+        orbit_direction=OrbitDirection.DESCENDING
+    )
+
+    DEM = DataSourceDefinition(
+        api_id='DEM',
+        wfs_id='DSS4',
+        source=_Source.DEM,
+        bands=_Bands.DEM,
+        is_timeless=True
+    )
+    MODIS = DataSourceDefinition(
+        api_id='MODIS',
+        wfs_id='DSS5',
+        service_url=ServiceUrl.USWEST,
+        source=_Source.MODIS,
+        processing_level=_ProcessingLevel.MCD43A4,
+        bands=_Bands.MODIS
+    )
+    LANDSAT8 = DataSourceDefinition(
+        api_id='L8L1C',
+        wfs_id='DSS6',
+        service_url=ServiceUrl.USWEST,
+        source=_Source.LANDSAT8,
+        processing_level=_ProcessingLevel.L1C,
+        bands=_Bands.LANDSAT8
+    )
+
+    SENTINEL5P = DataSourceDefinition(
+        api_id='S5PL2',
+        wfs_id='DSS7',
+        service_url=ServiceUrl.CREODIAS,
+        source=_Source.SENTINEL5P,
+        sensor_type=_SensorType.TROPOMI,
+        processing_level=_ProcessingLevel.L2,
+        bands=_Bands.SENTINEL5P
+    )
+    SENTINEL3_OLCI = DataSourceDefinition(
+        api_id='S3OLCI',
+        wfs_id='DSS8',
+        service_url=ServiceUrl.CREODIAS,
+        source=_Source.SENTINEL3,
+        sensor_type=_SensorType.OLCI,
+        processing_level=_ProcessingLevel.L1B,
+        bands=_Bands.SENTINEL3_OLCI
+    )
+    SENTINEL3_SLSTR = DataSourceDefinition(
+        api_id='S3SLSTR',
+        wfs_id='DSS9',
+        service_url=ServiceUrl.CREODIAS,
+        source=_Source.SENTINEL3,
+        sensor_type=_SensorType.SLSTR,
+        processing_level=_ProcessingLevel.L1B,
+        bands=_Bands.SENTINEL3_SLSTR
+    )
+
+    # EOCloud sources (which are only available on a development eocloud service):
+    LANDSAT5 = DataSourceDefinition(
+        wfs_id='L5.TILE',
+        service_url=ServiceUrl.EOCLOUD,
+        processing_level=_ProcessingLevel.GRD
+    )
+    LANDSAT7 = DataSourceDefinition(
+        wfs_id='L7.TILE',
+        service_url=ServiceUrl.EOCLOUD,
+        processing_level=_ProcessingLevel.GRD
+    )
+    ENVISAT_MERIS = DataSourceDefinition(
+        wfs_id='ENV.TILE',
+        service_url=ServiceUrl.EOCLOUD,
+        source=_Source.ENVISAT_MERIS
+    )
+
+    @classmethod
+    def define(cls, name, *, api_id=None, wfs_id=None, service_url=None, source=None, sensor_type=None,
```

Comment:
There are some additional collection parameters, e.g. view for [S3 SLSTR](https://docs.sentinel-hub.com/api/latest/data/sentinel-3-slstr-l1b/#view). I understand these parameters are "filters", but so are other parameters, that have their own argument here. 

Should the `define` also have an additional `**kwargs` parameter? (Despite how opposed I am to `**kwargs`, in this particular instance it might make sense)

Reply:
So far we have only parameters that are actually used somewhere in the code. So even if users would be able to add their own parameters these parameters would have no effect in the current code.